### PR TITLE
Bug Fix for Viewing the PSPublicIpAddress & PSDdosProtectionPlan objects

### PR DESCRIPTION
**Pull request by mchakka**
## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by Nickcandy** [comment]
Looks good to me. Please help me double-check @BethanyZhou

File: src/Network/Network/Models/PSDdosProtectionPlan.cs [hunk 1]
```diff
@@ -26,7 +26,7 @@ public class PSDdosProtectionPlan : PSTopLevelResource
 
         public List<PSResourceId> VirtualNetworks { get; set; }
 
-        public List<PSResourceId> PublicIpAddresses { get; set; }
+        public List<PSResourceId> PublicIPAddresses { get; set; }
```
Comment (BethanyZhou) [inline src/Network/Network/Models/PSDdosProtectionPlan.cs:29]: Renaming PublicI**p**Addresses to PublicI**P**Addresses is not allowed in regular release, which is regarded as a breaking change. Please revert this change.

File: src/Network/Network/Network.format.ps1xml [hunk 2]
```diff
@@ -1570,6 +1570,10 @@
                 <Label>DnsSettings</Label>
                 <PropertyName>DnsSettingsText</PropertyName>
               </ListItem>
+			  <ListItem>
+				<Label>DdosSettings</Label>
+				<PropertyName>DdosSettingsText</PropertyName>
+			  </ListItem>
```
Comment (BethanyZhou) [inline src/Network/Network/Network.format.ps1xml:1576]: Please correct indents

File: src/Network/Network/Network.format.ps1xml [hunk 3]
```diff
@@ -4081,6 +4085,10 @@
                 <Label>VirtualNetworks</Label>
                 <PropertyName>VirtualNetworksText</PropertyName>
               </ListItem>
+			  <ListItem>
+				<Label>PublicIPAddresses</Label>
+				<PropertyName>PublicIPAddressesText</PropertyName>
+			  </ListItem>
```
Comment (BethanyZhou) [inline src/Network/Network/Network.format.ps1xml:4091]: Please correct indents

File: src/Network/Network/Models/PSDdosProtectionPlan.cs (hunk 1 above)
Comment (mchakka) [reply src/Network/Network/Models/PSDdosProtectionPlan.cs:29]: we can wait till the next breaking change release for this PR - we want to rename this property to match the swagger change we made in this swagger PR.  https://github.com/Azure/azure-rest-api-specs/pull/21300 - so it's more of  abug fix rather than a breaking change since it never worked in the first place

Comment (BethanyZhou) [reply src/Network/Network/Models/PSDdosProtectionPlan.cs:29]: We have workaround to avoid breaking change for this case if you want to map PublicIPAddresses in swagger to PublicIpAddresses in PSDdosProtecyionPlan, so that this property would be work after then. Here is an example:
https://github.com/Azure/azure-powershell/blob/534cdbb7fb9c6f05c089e7ffd6c5e425abd69eb2/src/Network/Network/Common/NetworkResourceManagerProfile.cs#L1446-L1449

Also we are fine if you are good to wait next breaking change.

**Review by BethanyZhou — CHANGES_REQUESTED** [review]
Please fix other comments to get this PR ready for next breaking change release.

**Comment by BethanyZhou** [comment]
Hi @mchakka  this renaming can be regarded as a bug fix after I confirmed with team member. please fix other comments.

**Comment by mchakka** [comment]
Hey @BethanyZhou sorry for the delay, I have fixed the other comments :) PR should be good now - if you can merge it asap, that would be appreciated! 

Thanks again for all your help :)

File: src/Network/Network/ChangeLog.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<!--
+    <!--
```
Comment (BethanyZhou) [inline src/Network/Network/ChangeLog.md:1]:
```suggestion
<!--
```

File: src/Network/Network/Network.format.ps1xml [hunk 5]
```diff
@@ -983,15 +983,15 @@
...
                 <PropertyName>NatRulesText</PropertyName>
               </ListItem>
-			  <ListItem>
-				<Label>ExtendedLocation</Label>
-				<PropertyName>ExtendedLocationText</PropertyName>
-			  </ListItem>
+        <ListItem>
+        <Label>ExtendedLocation</Label>
+        <PropertyName>ExtendedLocationText</PropertyName>
+        </ListItem>
```
Comment (BethanyZhou) [inline src/Network/Network/Network.format.ps1xml:994]:
```suggestion
              <ListItem>
                <Label>ExtendedLocation</Label>
                <PropertyName>ExtendedLocationText</PropertyName>
              </ListItem>
```

File: src/Network/Network/Network.format.ps1xml [hunk 6]
```diff
@@ -5361,19 +5369,19 @@
...
+        <ListItem>
+        <Label>IpConfigurations</Label>
+        <PropertyName>IpConfigurationsText</PropertyName>
+        </ListItem>
+        <ListItem>
+        <Label>CustomNetworkInterfaceName</Label>
+        <PropertyName>CustomNetworkInterfaceName</PropertyName>
+        </ListItem>
+      </ListItems>
```
Comment (BethanyZhou) [inline src/Network/Network/Network.format.ps1xml:5384]:
```suggestion
              <ListItem>
                <Label>ApplicationSecurityGroups</Label>
                <PropertyName>ApplicationSecurityGroupsText</PropertyName>
              </ListItem>
              <ListItem>
                <Label>IpConfigurations</Label>
                <PropertyName>IpConfigurationsText</PropertyName>
              </ListItem>
              <ListItem>
                <Label>CustomNetworkInterfaceName</Label>
                <PropertyName>CustomNetworkInterfaceName</PropertyName>
              </ListItem>
            </ListItems>
```

**Review by BethanyZhou — DISMISSED** [review]
LGTM

**Comment by msJinLei** [comment]
```
2022-12-06T03:03:32.4481796Z   D:\a\_work\1\s\artifacts//StaticAnalysisResults\BreakingChangeIssues.csv Errors
2022-12-06T03:03:32.4483341Z   "Module","ClassName","Target","Severity","ProblemId","Description","Remediation"
2022-12-06T03:03:32.4484747Z   "Az.Network","Microsoft.Azure.Commands.Network.Automation.GetAzureRmDdosProtectionPlan","Get-AzDdosProtectionPlan","0","3010","The property 'PublicIpAddresses' of type 'Microsoft.Azure.Commands.Network.Models.PSDdosProtectionPlan' has been removed.","Add the property 'PublicIpAddresses' back to type 'Microsoft.Azure.Commands.Network.Models.PSDdosProtectionPlan'."
2022-12-06T03:03:32.4487730Z   "Az.Network","Microsoft.Azure.Commands.Network.Automation.GetAzureRmDdosProtectionPlan","Get-AzDdosProtectionPlan","0","3010","The property 'PublicIpAddressesText' of type 'Microsoft.Azure.Commands.Network.Models.PSDdosProtectionPlan' has been removed.","Add the property 'PublicIpAddressesText' back to type 'Microsoft.Azure.Commands.Network.Models.PSDdosProtectionPlan'."
2022-12-06T03:03:32.4490170Z   "Az.Network","Microsoft.Azure.Commands.Network.NewAzureRmDdosProtectionPlan","New-AzDdosProtectionPlan","0","3010","The property 'PublicIpAddresses' of type 'Microsoft.Azure.Commands.Network.Models.PSDdosProtectionPlan' has been removed.","Add the property 'PublicIpAddresses' back to type 'Microsoft.Azure.Commands.Network.Models.PSDdosProtectionPlan'."
2022-12-06T03:03:32.4492136Z   "Az.Network","Microsoft.Azure.Commands.Network.NewAzureRmDdosProtectionPlan","New-AzDdosProtectionPlan","0","3010","The property 'PublicIpAddressesText' of type 'Microsoft.Azure.Commands.Network.Models.PSDdosProtectionPlan' has been removed.","Add the property 'PublicIpAddressesText' back to type 'Microsoft.Azure.Commands.Network.Models.PSDdosProtectionPlan'."
2022-12-06T03:03:32.4493295Z
```

**Comment by mchakka** [comment]
@BethanyZhou can you help override the error message that @msJinLei posted - seems like the error is coming due to us renaming the property to fix the existing bug :(

**Comment by msJinLei** [comment]
> @BethanyZhou can you help override the error message that @msJinLei posted - seems like the error is coming due to us renaming the property to fix the existing bug :(

We are discussing on your PR's breaking changes now and will let you know how to fix later

**Comment by msJinLei** [comment]
@mchakka
Please refer to the https://github.com/Azure/azure-powershell/blob/main/documentation/Debugging-StaticAnalysis-Errors.md#breaking-changes to  suppress the breaking change error

**Review by msJinLei — APPROVED** [review]
